app/scripts/views/force_auth.js

```diff
@@ -38,12 +46,30 @@ define(function (require, exports, module) {
 
     _fatalError: null,
 
-    beforeRender: function () {
-      var self = this;
+    _getAndValidateAccountData: function () {
+      var fieldsToPick = ['email', 'uid'];
+      var accountData = {};
       var relier = this.relier;
 
-      if (! relier.has('email')) {
-        this._fatalError = AuthErrors.toError('FORCE_AUTH_EMAIL_REQUIRED');
+      fieldsToPick.forEach(function (fieldName) {
+        if (relier.has(fieldName)) {
+          accountData[fieldName] = relier.get(fieldName);
+        }
+      });
+
+      try {
+        return Transform.transformUsingSchema(
+            accountData, RELIER_DATA_SCHEMA, AuthErrors);
+      } catch (err) {
+        var errorPageUrl = getErrorPageUrl(err, this.translator);
+        this.window.location.href = errorPageUrl;
```

Comment:
I'm not fully convinced view's should handle their own redirection, but I already rat-holed enough.